lib/elixir/test/elixir/typespec_test.exs

```diff
@@ -1183,6 +1183,69 @@ defmodule TypespecTest do
       end)
     end
 
+    test "spec_to_quoted with maps with __struct__ key" do
+      defmodule A do
+        defstruct [:key]
+      end
+
+      defmodule B do
+        defstruct [:key]
+      end
+
+      bytecode =
+        test_module do
+          @spec single_struct(%A{}) :: {:ok}
+          def single_struct(arg), do: {:ok, arg}
+
+          @spec single_struct_key(%{__struct__: A}) :: {:ok}
+          def single_struct_key(arg), do: {:ok, arg}
+
+          @spec single_struct_key_type(%{__struct__: atom()}) :: {:ok}
+          def single_struct_key_type(arg), do: {:ok, arg}
+
+          @spec union_struct(%A{} | %B{}) :: {:ok}
+          def union_struct(arg), do: {:ok, arg}
+
+          @spec union_struct_key(%{__struct__: A | B}) :: {:ok}
+          def union_struct_key(arg), do: {:ok, arg}
+
+          @spec union_struct_key_type(%{__struct__: atom() | A | binary()}) :: {:ok}
```

Comment:
Please convert these `{:ok}` into `:ok`. As far as Elixir is concerned, we never use `{:ok}`. :) Thanks!